clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -6858,6 +6873,14 @@ private static <T> void assertCollectionIs(Collection<T> collection, T... elemen
         assertEquals(elements.length, collection.size(), "There are unexpected extra elements in the collection.");
     }
 
+    private <T> void waitUntilFutureComplete(Time time,  KafkaFuture<T> future) throws InterruptedException {
+        long waitMs = 10L;
+        TestUtils.retryOnExceptionWithTimeout(10000, 1, () -> {
```

Comment:
The previous code would fail fast if there was a timeout failure on `get()`? We're now retrying up to 10 seconds. Is this to allow extra time for the backoff or was the original test code sleeping unnecessarily long to begin with? Or is there a completely different reason?

Reply:
Well, it would be OK to wait until the backoff upper bound and then fail fast. Of course, what it's testing is that the gap between a pair of jittered delays is within the acceptable range, so this loop is just one way of doing a bounded wait.